tsdb/engine/tsm1/array_cursor.gen.go.tmpl

```diff
@@ -192,6 +192,8 @@ func (c *{{$type}}) reset(seek, end int64, cacheValues Values, tsmKeyCursor *Key
 		})
 		if c.cache.pos == len(c.cache.values) {
 			c.cache.pos--
+        } else if c.cache.values[c.cache.pos].UnixNano() == seek {
+            c.cache.pos--
```

Comment:
DescendingArrayCursors need to exclude points that fall on the right side of the range boundary.

Reply:
This is a port of a bug fix that you already fixed elsewhere, correct? Do we have a regression test here on this branch?

Reply:
Ah, I see lots of tests below.

Reply:
Yes we had a launcher test that failed when it was unskipped. That test now passes and I brought over the specific unit tests for it as well.

Reply:
Correction, this part of the file was not part of the original fix. These two lines do not appear to be in cloud, and so it's not clear to me whether this regression exists there as well. The launcher test that caught it now only lives in OSS.